File: Fortran/Lagrange Interpolation.f90
program LagrangeInterpolation
implicit none
integer::n,i,j,k
real,allocatable::x(:),l1(:),y(:)
real::x1,l=0
open(1,file='input.dat')
open(2,file='output.dat')
print*,"input the amount of interpolation data"
read(1,*)n
print*,n
allocate(x(n),l1(n),y(n))
l1=1
write(*,'("input the interpolation data:x,y")')
do i=1,n
  read(1,*),x(i),y(i)
  write(*,'("x=",f,"y=",f)')x(i),y(i)
end do
print*,"input x"
read(1,*)x1
write(*,'("x=",f)')x1
do i=1,n
  do j=1,n
    if(i==j)then
   continue
 else
   l1(i)=l1(i)*(x1-x(j))/(x(i)-x(j))
 end if
 end do
  l1(i)=l1(i)*y(i)
  l=l+l1(i)
end do
write(2,'("the interpolation result is:",f)')l
write(*,'("the interpolation result is:",f)')l
close(1)
close(2)
end

File: C/Lagrange Interpolation.c
#include<stdio.h>
#include<stdlib.h>
main()
{
 int n,i,j;
 float*x,*y,*l1,s=0,x1;
 FILE*fp;
 fp=fopen("data.dat","r");
 printf("please input the amount of datas\n");
 fscanf(fp,"%d",&n); printf("n=%d\n",n);
 printf("input the interpolation data:x,y\n");
 x=(float*)calloc(n,sizeof(float));
 y=(float*)calloc(n,sizeof(float));
 l1=(float*)calloc(n,sizeof(float));
 for(i=0;i<n;i++)
   fscanf(fp,"%f,%f",&x[i],&y[i]), printf("x=%f, y=%f\n",x[i],y[i]),l1[i]=1.0;
 printf("input x\n"); fscanf(fp,"%f",&x1); fclose(fp); printf("x=%f\n",x1);
 for(i=0;i<n;i++)
 {
  for(j=0;j<n;j++)
  {
   if(i==j) 
    continue;
   else
    l1[i]=l1[i]*(x1-x[j])/(x[i]-x[j]);
  }
  s+=l1[i]*y[i];
 }
 fp=fopen("output.dat","w");
 printf("the interpolation result is %f\n",s);
 fprintf(fp,"the interpolation result is %f",s);
 fclose(fp);
}